src/GraphQL.Tests/Types/InputObjectGraphTypeTests.cs

```diff
@@ -20,4 +23,53 @@ public void should_throw_an_exception_if_object_graph_type_contains_Input_object
         var exception = Should.Throw<ArgumentOutOfRangeException>(() => type.Field<InputObjectGraphType>("test"));
         exception.Message.ShouldContain("Output type 'Object' can have fields only of output types: ScalarGraphType, ObjectGraphType, InterfaceGraphType, UnionGraphType or EnumerationGraphType.");
     }
+
+    [Fact]
+    public async Task input_resolver_works()
+    {
+        // demonstrates having a StringGraphType field that accepts a Uri as input
+        // the string value is coerced to a Uri prior to beginning execution of the request
+        var inputType = new InputObjectGraphType<Class1>();
+        inputType.Field<StringGraphType, Uri>("url")
```

Comment:
Now here I was unable to write this:

```suggestion
        inputType.Field(x => x.Url)
            .Type<StringGraphType>()
```

since the `Type` method only accepted an implementation and not a type.  Perhaps an overload to the `Type` method should be added.  Notice that `TReturnType` will be `Uri` as that is what matches the type of the property, regardless of what data type is returned from the scalar.

Reply:
I should have written `inputType.Field(x => x.Url, type: typeof(StringGraphType))`